app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt

```diff
@@ -14,15 +11,13 @@ class LoginDataSource {
     private val auth = MainActivity.getAuth()
     private val dbRef = MainActivity.getDatabase().reference
 
-    fun login(userEmail: String, password: String, profile: User): Result<LoggedInUser> {
-        val authResultFuture = CompletableFuture<Result<LoggedInUser>>()
+    fun login(userEmail: String, password: String, profile: User): Result<User> {
+        val authResultFuture = CompletableFuture<Result<User>>()
 
         auth.signInWithEmailAndPassword(userEmail, password)
             .addOnCompleteListener { authTask ->
                 if (authTask.isSuccessful) {
-                    val loggedInUser =
-                        LoggedInUser(profile.displayName, profile.email, profile.role)
-                    authResultFuture.complete(Result.Success(loggedInUser))
+                    authResultFuture.complete(Result.Success(profile))
```

Comment:
So there is no more difference between a logged in user and other users, but at least it make this method straight forward.

Reply:
Yes not a difference at the Type level, but as finally it is also a `User` instance : in the `UserViewModel`, we have the information by variable naming that the `User` instance stored in is the `loggedInUser`.